### ISSUE TITLE
Getting boot loop errors

### ISSUE THREAD

**Issue by hammondses** [Skaronator/ESP8266-Load-Cell#6]
Hi there,

I'm attempting to get my Wemos D1 Mini Pro working with your bed sensor scripts, however after flashing both files I'm getting reboot errors. 

I modified config.h to suit my config, should I try it with yours?

I'm really new to the whole ESP8266 thing so please be gentle :)

I have wired it all correctly (I think) SCL -> SCK, SDA -> DT, 5v -> VCC & GND -> GND from ESP to HX711. I followed the diagram provided for the weight sensors to be wired to eachother and back to the HX711.

In the Serial Monitor I get total garbage at 9600 baud, but at 115200 I get this:

```
ets Jan  8 2013,rst cause:2, boot mode:(3,7)

load 0x4010f000, len 1384, room 16 
tail 8
chksum 0x2d
csum 0x2d
vbb28d4a3
~ld
```

Any help would be greatly appreciated!

**Comment by hammondses** [comment]
Update: Fixed my bootloop issues, issue was I needed to change the pins in the config, so now it boots but my monitor is full of absolute garbage no matter the baud I use :(

**Comment by Skaronator** [comment]
I'm actually not sure why the output is sometimes just garbage.

Did you install the Wemos Board Stuff? https://www.instructables.com/id/Programming-the-WeMos-Using-Arduino-SoftwareIDE/

**Comment by hammondses** [comment]
I think it's because it's a knock off from AliExpress, the WiFi test module seems to do nothing which doesn't look like a good sign from what I've been reading

**Comment by Skaronator** [comment]
Could you make a picture of it and maybe link the AliExpress product?

**Comment by hammondses** [comment]
https://s.click.aliexpress.com/e/b8QSpZQY

I'm not at home right now but I'll take a picture of it tomorrow when I'm home

**Comment by Skaronator** [comment]
Weird thats a LOLIN and not a Wemos. Mine looks completly different but just checked the Wemos page and it looks like they have a new version? https://wiki.wemos.cc/products:d1:d1_mini_pro

Mine looks like this: (Notice the ceramic antenna)

![image](https://user-images.githubusercontent.com/2886913/45870501-4b865a00-bd8b-11e8-8b48-b23ee77eaa03.png)

**Comment by hammondses** [comment]
I think the easiest thing for me to do is probably buy the same one you have, this is it? - https://www.trademe.co.nz/1773192104

**Comment by Skaronator** [comment]
Yes I think so but $14 is too expensive for that. This is what I've ordered earlier this year: https://www.aliexpress.com/item/WEMOS-D1-Mini-Pro-V1-0-16M-Bytes-External-Antenna-Connector-NodeMCU-ESP8266-ESP8266F-CP2104-WIFI/32861321679.html

(Just noticed that 1.0 not 1.1 but shouldn't matter)

**Comment by Skaronator** [comment]
Bascially it should be possible to run this on the v2 version but I don't have the hardware to validate this. Maybe I'll order some but it'll take almost a month until they are here.

**Comment by hammondses** [comment]
Ah, it's in NZD. The AliExpress link you gave is $11.77 so not too bad for it being local. I'll order one and let you know how I get on :)

**Comment by Skaronator** [comment]
Oh it's local? Didn't know that. The AliExpress link is $5 for me.
![image](https://user-images.githubusercontent.com/2886913/45871077-c603a980-bd8c-11e8-8846-b78b0006e416.png)

Edit: You probably meant total cost including shipping.

**Comment by hammondses** [comment]
Yeah, I did mean Inc shipping :)

I guess the issue too is that it's pretty difficult to find someone to test it who happens to have load sensors and the v2 hardware

**Comment by hammondses** [comment]
I'm now getting this error after fixing the pinouts in the code:

```
16:52:26.210 ->  ets Jan  8 2013,rst cause:2, boot mode:(3,6)
16:52:26.210 -> 
16:52:26.210 -> load 0x4010f000, len 1384, room 16 
16:52:26.210 -> tail 8
16:52:26.210 -> chksum 0x2d
16:52:26.210 -> csum 0x2d
16:52:26.255 -> vbb28d4a3
16:52:26.255 -> ~ld

```

**Comment by hammondses** [comment]
I decided to test the esp with a whole range of test sketches and it seems like all of the onboard functions are working and the HX711 calibration sketch gives me readouts of the load sensors.

After a bit of googling it seems it's quite a common error I'm getting when trying to use the hx711 and an esp8266, apparently putting in a yeild somewhere can fix it? 

See here - https://github.com/bogde/HX711/issues/67

**Comment by bachoo786** [comment]
@hammondses did you upload the code before connecting hx711 and the load cells?  Because I am uploading the code before attaching anything onto my esp8266 and I get errors as described in your first post.

Please help.

**Comment by hammondses** [comment]
I uploaded the code after connecting everything, which hardware revision are you using?

**Comment by bachoo786** [comment]
Sorry what do you mean by hardware revision ?

**Comment by hammondses** [comment]
Oh sorry I meant which esp do you have? I had the new version of the d1 mini pro which didn't seem to like the code so I'm waiting for the same revision to arrive which the creator has :)

**Comment by bachoo786** [comment]
I have the old version of d1 mini  and have the nodemcu none of them work I get the same error as described earlier.

**Comment by hammondses** [comment]
Hmm how odd, @Skaronator any suggestions?

**Comment by Skaronator** [comment]
Sorry, can't really help you since I'm relatively new to the whole Arduino stuff.

**Comment by hammondses** [comment]
@bachoo786 Which error are you seeing? If its the absolute garbage in the output monitor I fixed it on mine, I had to run a script to set the default baud rate of the ESP8266 to 9600 and did an AT+IPR=9600 command, then uploaded the sketch (I can't remember if it was sketch only or sketch+wifi) and it seems to have fixed it.

I gave up trying to get my version 2 board working, but the one that you have and the OP has I got working after setting the default baud rate of the board lower.

Let me know how you get on!

**Comment by hammondses** [comment]
@Skaronator what does your history graph for the sensor look like? For some reason i'm experiencing huge weight fluctuations randomly in the middle of the night or during the day for a few moments then back to normal. See attached picture

![image](https://user-images.githubusercontent.com/21191206/47076800-15d67300-d25c-11e8-8bb6-f2f0d225d89c.png)

**Comment by Skaronator** [comment]
@hammondses Mine looks like this.

![image](https://user-images.githubusercontent.com/2886913/47114056-0e629a00-d25b-11e8-9437-a15d56bf99ad.png)

The kg actually doesn't represent the true value (I'd need to change wiring to fix that) but it's enough to check if the bed is occupied or not. I use a binary_sensor to check this which is basically a template sensor that is true once it's above 50 kg.

![image](https://user-images.githubusercontent.com/2886913/47114186-67cac900-d25b-11e8-93f7-9e27e298372b.png)

**Comment by amotl** [comment]
Hi there, just wanted to let you know we are working on https://github.com/bogde/HX711/pull/123, which you might want to give a try. It is an attempt to remedy some of the issues of the Arduino HX711 library, especially when it came to ESP8266 and ESP32 support.

**Comment by sagartec** [comment]
@hammondses did you solve the reset problem. I am also facing the same issue now.

**Comment by hammondses** [comment]
Yeah I ended up changing to the exact same board as the original poster was using, also noticed some boards don't like having the letter in the pinout, ie D3 D2 I changed to 3 and 2 which resolved my issues with other projects.
